core/network/src/main/java/com/gkp/core/network/util/NetworkApiCall.kt

```diff
@@ -11,29 +11,30 @@ import java.nio.channels.UnresolvedAddressException
 import kotlin.coroutines.cancellation.CancellationException
 
 inline fun <T> networkApiCall(
-    crossinline apiCall: suspend () -> T
+    crossinline apiCall: suspend () -> T,
 ): Flow<TaskyResult<T>> = flow {
     emit(TaskyResult.Loading)
     val response = apiCall.invoke()
     emit(TaskyResult.Success(response))
 }.catch { cause: Throwable ->
-    if (cause is CancellationException) throw  cause
+    if (cause is CancellationException) throw cause
     emit(TaskyResult.Error(getErrorType(cause)))
 }
```

Comment:
The extra `Loading` emission doesn't really add any value here, since your VM can easily update the loading state without. When removing that, the `Flow` also becomes unnecessary, since you'll only have one return value from that point on, so this can just be a normal suspend function.
In the VM, you can then just do this:
```
viewModelScope.launch {
    state = state.copy(isLoading = true)
    // your use case/repository call
    state = state.copy(isLoading = false)
}
```